src/core/libraries/dialogs/ime_dialog_ui.cpp

```diff
@@ -0,0 +1,488 @@
+        iconv_close(utf8_to_orbis);
+    }
+#endif
+
+    if (title) {
+        delete[] title;
+    }
+
+    if (placeholder) {
+        delete[] placeholder;
+    }
+}
+
+bool ImeDialogState::CallKeyboardFilter(const OrbisImeKeycode* src_keycode, u16* out_keycode,
+                                        u32* out_status) {
+    if (!keyboard_filter) {
+        return true;
+    }
+
+    auto* linker = Common::Singleton<Core::Linker>::Instance();
+    int ret = linker->ExecuteGuest(keyboard_filter, src_keycode, out_keycode, out_status, nullptr);
+
+    return ret == 0;
+}
+
+bool ImeDialogState::ConvertOrbisToUTF8(const char16_t* orbis_text, std::size_t orbis_text_len,
+                                        char* utf8_text, std::size_t utf8_text_len) {
+    std::fill(utf8_text, utf8_text + utf8_text_len, '\0');
+#ifndef _WIN32
+    std::size_t orbis_text_len_bytes = orbis_text_len * sizeof(char16_t);
+    std::size_t utf8_text_len_bytes = utf8_text_len * sizeof(char);
+
+    char16_t* orbis_text_ptr = const_cast<char16_t*>(orbis_text);
+    char* utf8_text_ptr = utf8_text;
+
+    std::size_t result = iconv(orbis_to_utf8, (char**)&orbis_text_ptr, &orbis_text_len_bytes,
+                               (char**)&utf8_text_ptr, &utf8_text_len_bytes);
+
+    return result != (std::size_t)-1;
+#else
+    int required_size =
+        WideCharToMultiByte(CP_UTF8, 0, reinterpret_cast<const wchar_t*>(orbis_text),
+                            orbis_text_len, nullptr, 0, nullptr, nullptr);
+    if (required_size > utf8_text_len) {
+        return false;
+    }
+
+    int converted_size =
+        WideCharToMultiByte(CP_UTF8, 0, reinterpret_cast<const wchar_t*>(orbis_text),
+                            orbis_text_len, utf8_text, utf8_text_len, nullptr, nullptr);
+
+    return converted_size != 0;
+#endif
+}
+
+bool ImeDialogState::ConvertUTF8ToOrbis(const char* utf8_text, std::size_t utf8_text_len,
+                                        char16_t* orbis_text, std::size_t orbis_text_len) {
+    std::fill(orbis_text, orbis_text + orbis_text_len, u'\0');
+#ifndef _WIN32
+    std::size_t utf8_text_len_bytes = utf8_text_len * sizeof(char);
+    std::size_t orbis_text_len_bytes = orbis_text_len * sizeof(char16_t);
+
+    char* utf8_text_ptr = const_cast<char*>(utf8_text);
+    char16_t* orbis_text_ptr = orbis_text;
+
+    std::size_t result = iconv(utf8_to_orbis, (char**)&utf8_text_ptr, &utf8_text_len_bytes,
+                               (char**)&orbis_text_ptr, &orbis_text_len_bytes);
+
+    return result != (std::size_t)-1;
+#else
+    int required_size = MultiByteToWideChar(CP_UTF8, 0, utf8_text, utf8_text_len, nullptr, 0);
+    if (required_size > orbis_text_len) {
+        return false;
+    }
+
+    int converted_size =
+        MultiByteToWideChar(CP_UTF8, 0, utf8_text, utf8_text_len,
+                            reinterpret_cast<wchar_t*>(orbis_text), orbis_text_len);
+
+    return converted_size != 0;
+#endif
+}
```

Comment:
The ImGUI function convert from UTF8 to UTF32 but PS4 uses UTF16. 
Wchar is different size on windows and linux/macos. We need a function to convert specificly from UTF8 to UTF16 and the other way arround.

Reply:
Remove the `IMGUI_USE_WCHAR32` from imgui_config.h

Reply:
I searched the imgui source code and, sadly, it only provides implementation for utf32. :(

I even found a coment in imgui that they stopped using ImTextStrFromUtf8 because they need to support both utf16 and 32.

https://github.com/shadps4-emu/ext-imgui/blob/636cd4a7d623a2bc9bf59bb3acbb4ca075befba3/imgui.cpp#L2268C1-L2272C81

Reply:
Yeah, but that definition in imgui_config.h controls whether ImWchar is 16 or 32 bits.
To ensure nobody changes, you can even do something like:
```Cpp
#ifdef IMGUI_USE_WCHAR32
#error "This project uses 16 bits wchar standard like Orbis"
#endif
```